interp/interp.go

```diff
@@ -0,0 +1,154 @@
+// Copyright ©2020 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package interp
+
+import (
+	"sort"
+)
+
+// Predictor predicts the value of a function. It handles both
+// interpolation and extrapolation.
+type Predictor interface {
+	// Predict returns the predicted value at x.
+	Predict(x float64) float64
+}
+
+// Fitter fits a predictor to data.
+type Fitter interface {
+	// Fit fits a predictor to (X, Y) value pairs provided as two slices.
+	// It panics if len(xs) < 2, elements of xs are not strictly increasing or len(xs) != len(ys).
+	Fit(xs, ys []float64)
```

Comment:
Sorry, last thing. This should return an error. We definitely want this interface to be able to support least squares regression, where the prediction could error.

Reply:
Good point.